tests/fixtures/data.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2022 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+from pathlib import Path
+
+from omegaconf import DictConfig
+from pytest import fixture
+from torch.utils.data import DataLoader
+from transformers import PreTrainedTokenizerBase
+
+from llmfoundry.data.finetuning.dataloader import build_finetuning_dataloader
+from tests.data_utils import make_tiny_ft_dataset
+
+
+@fixture
+def tiny_ft_dataset_path(tmp_path: Path, dataset_size: int = 4) -> Path:
```

Comment:
can we make this fixture clean up after itself?

Reply:
do you mean remove the dataset after the test is done? If so, I think the tmp_path fixture we are using will do that automatically

Reply:
oh yeah, good point. Can you go remove the place where I explicitly `rmtree` the dataset?

Reply:
yup, can do that! also we should add a check to only create the dataset on rank 0, right?

Reply:
added logic below to only create the dataset on rank 0 if worldsize != 1